clients/src/main/java/org/apache/kafka/clients/admin/internals/AdminMetadataManager.java

```diff
@@ -226,20 +225,30 @@ public void transitionToUpdatePending(long now) {
         this.lastMetadataFetchAttemptMs = now;
     }
 
+    public void updateFailed(Throwable exception) {
+        // We depend on pending calls to request another metadata update
+        this.state = State.QUIESCENT;
+
+        if (exception instanceof AuthenticationException) {
+            log.warn("Metadata update failed due to authentication error", exception);
+            this.authException = (AuthenticationException) exception;
+        } else {
+            log.debug("Metadata update failed", exception);
+        }
+    }
+
     /**
      * Receive new metadata, and transition into the QUIESCENT state.
      * Updates lastMetadataUpdateMs, cluster, and authException.
      */
-    public void update(Cluster cluster, long now, AuthenticationException authException) {
+    public void update(Cluster cluster, long now) {
```

Comment:
Should we reset `authException` to null in update()?

Reply:
Yes, good call. I noticed that earlier, but forgot about it. I will add a test case.